spring-integration-core/src/main/java/org/springframework/integration/support/management/graph/IntegrationGraphServer.java

```diff
@@ -79,6 +81,17 @@ public void onApplicationEvent(ContextRefreshedEvent event) {
 	}
 
 	private synchronized Graph buildGraph() {
+		String implementationVersion = IntegrationGraphServer.class.getPackage().getImplementationVersion();
+		if (implementationVersion == null) {
+			implementationVersion = "unknown - is Spring Integration running from the distribution jar?";
+		}
+		Map<String, Object> descriptor = new HashMap<String, Object>();
+		descriptor.put("provider", "spring-integration");
+		descriptor.put("providerVersion", implementationVersion);
+		descriptor.put("providerFormatVersion", GRAPH_VERSION);
+		if (this.applicationContext.containsBean("applicationNameAndVersion")) {
```

Comment:
Yeah... I worried here about Boot's:
```
${spring.application.name}
${application.version}
```
That's why I don't see reason in the separate extra bean, when it really would be better to free ourselves via external injections.
And yes: get a gain, finally, from the Boot environment.